### PR TITLE
[EMB-510][Collections] Add submission confirmation modal

### PR DESCRIPTION
**Pull request by adlius**
## Purpose

This PR does three things:
1. Add a `collection-submission-confirmation-modal` component for alerting users about project becoming public after adding to collection.
2. Fix a problem where `ember-onbeforeuload` prevents redirection to another page after successful submission/update of a project. (Not part of this ticket but related, sort of.)
3. Fix the blueprint so that generating a component for an in-repo-addon that uses engine would place the test files under `test/engines/<engine-name>/integration/components/<component-name>` and uses the correct setup hooks. (Not part of this ticket. Does not need QA.)
    - Also fix the unnecessary generation of `/app` for for engine components.

## Summary of Changes

The following changes are made, respectively:
- A new `collection-submission-confirmation-modal` is added.
  - Make use of the component in `collections-submission` component.
- Add `resetPageDirty` action to both the `submit` and `edit` controller to reset the `isPageDirty` property.
  - The action is then passed to the `collections-submission` component for consumption.
- Override the `filesPath()` and `files()` hook to make sure blueprint generation has the expected behavior. See code comments.

## QA Notes

The modal only shows up when submitting a project. It should not show up when updating.

## Ticket

https://openscience.atlassian.net/browse/EMB-510

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: lib/collections/addon/components/collection-submission-confirmation-modal/template.hbs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<BsModal 
+    @open={{@openModal}}
+    @onSubmit={{action @addToCollection}}
+    @onHidden={{action @cancel}}
+    as |modal|
+>
+    <modal.header>
+        <h3>{{t (concat this.i18nKeyPrefix 'header')}}</h3>
+    </modal.header>
+    <modal.body class=componentCssClassName>
```
Comment (jamescdavis) [inline lib/collections/addon/components/collection-submission-confirmation-modal/template.hbs]: I'm guessing you copied `delete-node-modal`, which, apparently, has some cruft left over from when we were using `ember-component-css`:
```suggestion
    <modal.body>
```

File: lib/collections/addon/components/collections-submission/component.ts [hunk 2]
```diff
@@ -135,38 +138,50 @@ export default class Submit extends Component {
...
     @action
-    reset(this: Submit) {
-        this.collectedMetadatum.rollbackAttributes();
-
-        this.setProperties({
-            activeSection: Section.project,
-            savedSections: [],
-        });
-
+    cancel(this: Submit) {
```
Comment (jamescdavis) [inline lib/collections/addon/components/collections-submission/component.ts]: Typing `this` is not required here.

**Comment by adlius** [comment]
@jamescdavis Okay, back to you first for a quick sanity check. Still need to add some blueprint tests.

File: blueprints/component-test/index.js [hunk 3]
```diff
@@ -48,9 +54,21 @@ module.exports = {
...
+    filesPath(options) {
+        const engineName = options.inRepoAddon;
+        const isEngine = fs.existsSync(path.join(options.project.root, 'lib', engineName, 'addon', 'engine.js'));
+        if (isEngine) {
+            // if the in-repo-addon within which the new component is added uses ember engine
+            // we tell ember-cli to use the dir structure and __test__.ts in 'engine-files'
+            return path.join(this.path, 'engine-files');
+        }
+        return path.join(this.path, 'files');
+    },
```
Comment (jamescdavis) [inline blueprints/component-test/index.js]: :100:

File: blueprints/component-addon/index.js [hunk 4]
```diff
@@ -37,4 +38,15 @@ module.exports = {
...
+        const engineName = this.options.inRepoAddon;
+        const isEngine = fs.existsSync(path.join(this.options.project.root, 'lib', engineName, 'addon', 'engine.js'));
+        if (isEngine) {
+            // if the in-repo-addon within which the new component is added uses ember engine
+            // we return an empty array
+            // so that the blueprint would not be picked up and used for generating unnecessary `/app` tree
+            return [];
+        }
+        return this._super();
+    },
```
Comment (jamescdavis) [inline blueprints/component-addon/index.js]: 🥇

**Pull request merged**